docs/conceptual/core/consensus.md

```diff
@@ -2,14 +2,104 @@
+
+DIAGRAM
+
+So when conflicting blocks (blocks that contain conflicting transactions) exist, there must be a fork. Each side of the fork represents a separate blockchain. Because no individual blockchain can contain conflicting transactions, we have reduced the problem of transaction conflict exclusion to choosing a fork in the chain.
+
+One necessary property in a fork choice rule is "liveness". A rule with liveness guarantees that the chain will keep growing. Consider the following fork choice rules as examples.
+
+### Longest Chain Rule
+First introduced in Bitcoin, the ubiquitous longest chain rule, simply says that the cannonical chain is the longest chain.
+
+DIAGRAM
+
+### GHOST Rule
+The Greedy Heaviest Observed SubTree rule says that, starting at the genesis block, each fork is resolved by choosing the branch that has the most blocks built on it recursively.
+
+DIAGRAM
+
+## Block Production
+Some nodes in a blockchain network are able to produce new blocks. Exactly which nodes may author blocks depends on which consensus engine you're using. In a centralized network, a single node might author all the blocks, whereas in a completely permissionless network, any node that wishes to, may produce a block.
+
+Nodes that are following the rules of the consensus protocol always produce new blocks on top of the chain that the fork choice rule tells them to. But blockchains still must tolerate nodes that cheat and build blocks in the wrong place. Considering the example fork choice rules we studied above, it's easy to see that an attacker who wanted to revert many blocks worth of transactions could do so by building a longer chain.
+
+To prevent malicious nodes from creating attack chains and overwhelming the honest chain whenever they please, block production must be throttled so that nodes can only produce blocks at a given rate. There are two common ways of achieving this throttling.
+
+### Proof of Work
+In proof of work systems like Bitcoin, any node may produce a block at any time, so long as they have solved a computationally-intensive problem. Solving the problem takes cpu time, and thus miners can only produce blocks in proportion with their computing resources. Substrate provides a Proof of Work consensus engine.
+
+### Slots
+When using a slot based consensus algorithm, there must be a known set of validators who are permitted to produce blocks. Time is divided up into discrete slots, and during each slot only some of the validators may produce a block. The specifics of which validators can author blocks during each slot vary from engine to engine. Substrate provides Aura and Babe, both of which are slot-based block authoring engines.
+
+## Finality
+Transactors in any system want to know when their transaction is finalized, and blockchain is no different. In some traditional systems, finality happens when a receipt is handed over, or papers are signed.
```

Comment:
`Transactors` looks not familiar to me, how about `users`?

Reply:
I'm open to that suggestion. Do others agree either way? I had originally chosen the term transactors to be as specific as possible (rule out those who use the blockchain for forensics, or only as full nodes, etc).

Reply:
Actually, I just took the suggestion. If anyone agrees with my reason for using "transactors" we can switch back. Until then, let's keep this moving.